### DISCUSSION TITLE
Dependence of tile resolution on the level of detail of the planet model

### DISCUSSION THREAD

**Discussion by slavkotm** [pelicanmapping/osgearth#2325] [Support Q&A] [answered]
![image](https://github.com/gwaldron/osgearth/assets/89938706/93a412f0-357e-4510-8716-68fd59d52b52)
![image](https://github.com/gwaldron/osgearth/assets/89938706/aab388b4-f7b2-4f1e-9ed3-cbdc3a57099e)
These pictures show a planet model with one large earth tile at level 0 (if I understand correctly, it's just one large earth texture stretched over the earth model) in normal mode, and in polygon mode. And the question is: is there any relationship between the level of detail of the earth model and the level of detail of the earth tiles themselves? Because including the polygonal mod, every time we approach the model of the earth, its detail is doubled, that is, the polygons are divided into two, and the size of the tiles, respectively, turns out to be different, unfortunately I can’t say exactly how it changes, but it looks like it’s different way. Here's what I'd like to understand: Is there a relationship between detail levels and tile detail? And is it possible to somehow control it in the code? Because, for example, by calling the method "getTileDimensions" we get the width and height of the tile at a certain zoom level, we can specify the width and height of the tile even up to centimeters, and the detail is limited already at a certain zoom level. I hope I have explained the essence of this topic clearly.

**Comment by gwaldron** [comment, marked as answer]
The image resolution is tied to the tile LOD. The imagery for each tile is (by default) 256x256 pixels. Sometimes you can change that by adjusting the `tile_size` property in the layer, but this only works for layer types (like `GDALImage`) that are not pre-tiled. The data in your screenshot appears to be from ReadyMap, which is pre-tiled to 256x256 pixels and you cannot change it.

If you want to more closely "match" the imagery resolution to the tile's pixel size on the screen, you can try this range mode:
```xml
<map>
   <options>
      <terrain range_mode="PIXEL_SIZE_ON_SCREEN"/>
...
```

**Reply by slavkotm** [reply]
Thanks for the answer! There is one more question: I have a gradient file .tif that I cut from the program qgis with a resolution of 0.000278 in latitude and longitude (this is approximately equal to 30 meters). As I also found out, the detail of each tile at each level is equal to 180, 90, 45, 22.5, etc., respectively, that is, each level of detail has a value equal to divided by two from the previous one. At detail level 19 and 20, there is a tile resolution of 0.000343323 (near 37 meters) and 0.0001716615 (near 18 meters), I found this using the getTileDimensions(levelLOD, out_width, out_height) function. Remembering the resolution of the .tif file where the resolution is 30 meters (0.000278 degrees) how can I make the level of detail more accurate - specifically suitable for my .tif file with my resolution of 30 meters? As I understand it, this cannot be done, since the program has already set the values from which the resolution of tiles begins at subsequent levels of detail - 180 and 180. Or is it possible to set the level of detail to 0.000278 value? The only way out for me is to just take this value of the resolution of my .tif file and just compare it with the detail levels until I find a suitable one to which I will process the detail, that is, level 20 with a tile resolution of 0.0001716615 (18 meters) will be suitable and I will set it as the maximum level of detail for my file. I am uploading a .tif file using GDALImageLayer

**Reply by gwaldron** [reply]
Sorry for the late reply. No, it's not possible to match it exactly. The PIXEL_SIZE_ON_SCREEN approach should get reasonable results, hopefully.
